src/bp/core/services/action/local-action-server.ts

```diff
@@ -0,0 +1,144 @@
+import bodyParser from 'body-parser'
+import { Logger } from 'botpress/sdk'
+import { ActionDefinition, ActionParameterDefinition } from 'common/typings'
+import { Config } from 'core/app'
+import { ConfigProvider } from 'core/config/config-loader'
+import { BadRequestError, UnauthorizedError } from 'core/routers/errors'
+import { AUDIENCE } from 'core/routers/sdk/utils'
+import { TYPES } from 'core/types'
+import express, { NextFunction, Request, Response } from 'express'
+import { inject, injectable, tagged } from 'inversify'
+import Joi from 'joi'
+import jsonwebtoken from 'jsonwebtoken'
+import _ from 'lodash'
+
+import { BotService } from '../bot-service'
+
+import ActionService from './action-service'
+import { HTTP_ACTIONS_PARAM_TYPES } from './utils'
+
+const _validateRunRequest = (botService: BotService) => async (req: Request, res: Response, next: NextFunction) => {
+  const { appSecret } = await Config.getBotpressConfig()
```

Comment:
I,m not sure how I could do that, since process.APP_SECRET is not available in this thread. It's only available in the web thread. Also, the `checkJwtSecret()` function has the side effect of writing to disk or the database, so I'm not sure how we could avoid a race condition between the web thread and the local action server thread for who's writing the app secret first.

I'm thinking that maybe the master thread should write the app secret to the Botpress Config file, and then the web thread and local action server thread should simply read the Botpress Config file to retrieve the app secret.

What do you think?

Reply:
Instead of starting the ACTION_SERVER worker directly at the startup in `cluster.ts`, maybe we could just spawn the web worker, and let the web worker ask the main process to start the action server. 

It could remove the need for the action server to read the config at all, and would not fork a process if there's no local server, ex:

```js
this.config = await this.configProvider.getBotpressConfig()
//...
    AppLifecycle.setDone(AppLifecycleEvents.CONFIGURATION_LOADED)

    await this.restoreDebugScope()
    await this.checkJwtSecret()

    const {enabled, port} = this.config.actionServers.localActionServer
    if(enabled){
      await cluster.startActionServer({ WORKER_TYPE: 'ACTION_WORKER', APP_SECRET: process.APP_SECRET, PORT: port })
    }
```

Reply:
This won't work, as `cluster.fork()` is only available from the master process
https://nodejs.org/api/cluster.html#cluster_cluster_fork_env

I think this could work if the Web process sent a message to the Master process, and the Master process creates the fork. I will test this.

Reply:
Done in 85c55a26c1926fecacb3434cb23c9e27121284f9